tensorflow/python/kernel_tests/math_ops/aggregate_ops_test.py

```diff
@@ -59,9 +59,16 @@ def testAddN(self):
         for count in range(1, self._MAX_N + 1):
           data = [self._buildData((2, 2), dtype) for _ in range(count)]
           actual = self.evaluate(math_ops.add_n(data))
-          expected = np.sum(np.vstack(
-              [np.expand_dims(d, 0) for d in data]), axis=0)
+          if dtype==dtypes.bfloat16:
+            data_fp32 = [elem.astype(np.float32) for elem in data]
```

Comment:
Why is it necessary to go through f32 here?  Is there too much loss of precision in the expected answer?

Reply:
In case of bfloat16 dtype, the Numpy function sum() has the following issue:
If the sum of inputs falls between 256 and 512, then the function only returns 256 but anything outside of this range works correctly. This can result in loss of precision and hence the workaround of going through fp32 is required.

Reply:
Your PR is behind - the tolerances have already been adjusted to account for float16 and bfloat16, and it passes for me without going through float32 for the expected sum.  Sync, and it should (hopefully) work by just adding `dtypes.bfloat16` to the list of supported types.

Reply:
Tested without the FP32 pass; it is working correctly and also updated the the tolerances.